components/tikv_alloc/src/default.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(dead_code)]
+
```

Comment:
We shouldn't need this

Reply:
@nrc   Because I don't know how to express cfg, to avoid `unused` errors in CI test.

like this:

```
warning: function is never used: `dump_stats`
 --> components/tikv_alloc/src/default.rs:6:1
  |
6 | pub fn dump_stats() -> String {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: #[warn(dead_code)] on by default

warning: function is never used: `dump_prof`
 --> components/tikv_alloc/src/default.rs:9:1
  |
9 | pub fn dump_prof(_path: Option<&str>) {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: function is never used: `fetch_stats`
  --> components/tikv_alloc/src/default.rs:11:1
   |
11 | pub fn fetch_stats() -> io::Result<Option<AllocStats>> {
``` 

I want to express this meaning by cfg, but it's wrong.

```rust
#[cfg(all(any(windows,fuzzing, unix), not(feature = "jemalloc")))]
mod default;
```

Reply:
I don't have opinions about this. The cfg logic in this crate is brutal so I'm happy for whatever works for now.

Reply:
One way to avoid this is to get rid of `default.rs` and duplicate the few defs in there in the allocator-specific files. Or to `mod default` directly from the allocator-specific files instead of the crate root.

Reply:
I thought about ways to add redundant code to solve it. But it doesn't feel very good, maybe this place can wait until later to refactor.

Reply:
I'm fine to leave it till later, it does seem to suggest we could do things better some how

Reply:
Added FIXME comment

```rust
// FIXME: https://github.com/tikv/tikv/pull/4524#discussion_r282754034 
#![allow(dead_code)]
```